bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.java

```diff
@@ -1,460 +0,0 @@
-package com.bugsnag.android.ndk;
-
-import com.bugsnag.android.Breadcrumb;
-import com.bugsnag.android.Configuration;
-import com.bugsnag.android.NativeInterface;
-
-import android.os.Build;
-import android.util.Log;
-import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
-
-import java.io.File;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.util.Date;
-import java.util.List;
-import java.util.Locale;
-import java.util.Observable;
-import java.util.Observer;
-import java.util.TimeZone;
-import java.util.UUID;
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
-
-/**
- * Observes changes in the Bugsnag environment, propagating them to the native layer
- */
-public class NativeBridge implements Observer {
-
-    private static final int METADATA_SECTION = 0;
-    private static final int METADATA_KEY = 1;
-    private static final int METADATA_VALUE = 2;
-    private static final Lock lock = new ReentrantLock();
-    private static final AtomicBoolean installed = new AtomicBoolean(false);
-
-    public static native void install(@NonNull String reportingDirectory,
-                                      boolean autoDetectNdkCrashes,
-                                      int apiLevel, boolean is32bit);
-
-    public static native void deliverReportAtPath(@NonNull String filePath);
-
-    public static native void addBreadcrumb(@NonNull String name, @NonNull String type,
-                                            @NonNull String timestamp, @NonNull Object metadata);
-
-    public static native void addMetadataString(@NonNull String tab, @NonNull String key,
-                                                @NonNull String value);
-
-    public static native void addMetadataDouble(@NonNull String tab, @NonNull String key,
-                                                double value);
-
-    public static native void addMetadataBoolean(@NonNull String tab, @NonNull String key,
-                                                 boolean value);
-
-    public static native void addHandledEvent();
-
-    public static native void addUnhandledEvent();
-
-    public static native void clearMetadataTab(@NonNull String tab);
-
-    public static native void removeMetadata(@NonNull String tab,@NonNull  String key);
-
-    public static native void startedSession(@NonNull String sessionID, @NonNull String key,
-                                             int handledCount, int unhandledCount);
-
-    public static native void pausedSession();
-
-    public static native void updateAppVersion(@NonNull String appVersion);
```

Comment:
Is this now completely unused? (Along with others that aren't in the refactor?) Should we remove the `MessageType` in the native interface?

Reply:
Yep this is unused - I've updated `MessageType` to remove messages which aren't currently used, and the rest will be removed in separate PRs which replaces them with `StateEvent`.